### ISSUE TITLE
Concrete paths like /pets/mine should be matched before templated paths like /pets/{petId}

### ISSUE THREAD

**Issue by andersk** [p1c2u/openapi-core#265]
According to the [specification](https://swagger.io/specification/#paths-object):

> When matching URLs, concrete (non-templated) paths would be matched before their templated counterparts.
> […]
> Assuming the following paths, the concrete definition, `/pets/mine`, will be matched first if used:
> ```
>   /pets/{petId}
>   /pets/mine
> ```

`openapi_core` fails to obey this, as demonstrated by the following test, which incorrectly raises `openapi_core.casting.schemas.exceptions.CastError: Failed to cast value mine to type integer`. The test passes if `/pets/mine` is moved before `/pets/{petId}` in the `paths` dict.

```python
from openapi_core import create_spec
from openapi_core.testing import MockRequest
from openapi_core.validation.request.validators import RequestValidator

spec = create_spec(
    {
        "openapi": "3.0.0",
        "info": {"title": "test", "version": "0"},
        "paths": {
            "/pets/{petId}": {
                "get": {
                    "parameters": [
                        {
                            "name": "petId",
                            "in": "path",
                            "required": True,
                            "schema": {"type": "integer"},
                        },
                    ],
                    "responses": {"200": {"description": "ok"}},
                },
            },
            "/pets/mine": {"get": {"responses": {"200": {"description": "ok"}}}},
        },
    },
)
validator = RequestValidator(spec)
request = MockRequest("http://localhost", "GET", "/pets/mine")
result = validator.validate(request)
result.raise_for_errors()
```

**Comment by stojan-jovic** [comment]
Indeed, this is a bug in the latest version again. In the previous version (i.e. `0.13.3`) there was also issue with resolving paths, but not only the similar ones like here.

@p1c2u Is this related to previous fix for paths resolving? I assume that solution would be simple to cover also this use case.

**Comment by p1c2u** [comment]
It was fixed in 0.13.5. Closing
